libbz2-rs-sys/src/decompress.rs

```diff
@@ -768,12 +764,11 @@ pub(crate) fn decompress(
                     if zn > 20 {
                         error!(BZ_DATA_ERROR);
                     } else if zvec <= s.limit[gLimit as usize][zn as usize] {
-                        if !(0..258).contains(&(zvec - s.base[gBase as usize][zn as usize])) {
-                            error!(BZ_DATA_ERROR);
-                        } else {
-                            nextSym = s.perm[gPerm as usize]
-                                [(zvec - s.base[gBase as usize][zn as usize]) as usize];
-                        }
+                        let index = zvec - s.base[gBase as usize][zn as usize];
+                        nextSym = match s.perm[gPerm as usize].get(index as usize) {
+                            Some(&nextSym) => nextSym,
+                            None => error!(BZ_DATA_ERROR),
+                        };
```

Comment:
the `s.perm[i]` arrays are `258` elements long, so the check for the `index` value to be in bounds was happening twice (though of course llvm might optimize that out)